src/pages/search/models.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  Container,
+  Title,
+  Stack,
+  SegmentedControl,
+  SegmentedControlItem,
+  Group,
+  ThemeIcon,
+  Text,
+  createStyles,
+  Box,
+  Center,
+  Loader,
+} from '@mantine/core';
+import {
+  InstantSearch,
+  SearchBox,
+  useInfiniteHits,
+  useInstantSearch,
+  useSearchBox,
+} from 'react-instantsearch-hooks-web';
+import { instantMeiliSearch } from '@meilisearch/instant-meilisearch';
+
+import { env } from '~/env/client.mjs';
+import {
+  ChipRefinementList,
+  SearchableMultiSelectRefinementList,
+  SortBy,
+} from '~/components/Search/CustomSearchComponents';
+import { routing } from '~/components/Search/useSearchState';
+import { useInView } from 'react-intersection-observer';
+import { useEffect } from 'react';
+import { ModelGetAll } from '~/types/router';
+import { ModelCard } from '~/components/Cards/ModelCard';
+import { SearchHeader } from '~/components/Search/SearchHeader';
+
+const searchClient = instantMeiliSearch(
+  env.NEXT_PUBLIC_SEARCH_HOST as string,
+  env.NEXT_PUBLIC_SEARCH_CLIENT_KEY,
+  { primaryKey: 'id', keepZeroFacets: true }
+);
+
+export default function Search() {
+  return (
+    <InstantSearch searchClient={searchClient} indexName="models" routing={routing}>
+      <Container fluid>
+        <Stack
+          sx={(theme) => ({
+            height: 'calc(100vh - 2 * var(--mantine-header-height,50px))',
+            position: 'fixed',
+            left: 0,
+            top: 'var(--mantine-header-height,50px)',
+            width: '377px',
+            overflowY: 'auto',
+            padding: theme.spacing.md,
+          })}
+        >
+          <RenderFilters />
+        </Stack>
+
+        <Stack pl={377} w="100%">
+          <SearchHeader />
+          <ModelsHitList />
+        </Stack>
+      </Container>
+    </InstantSearch>
+  );
+}
+
+const RenderFilters = () => {
+  return (
+    <>
+      <SortBy
+        title="Sort models by"
+        items={[
+          { label: 'Highest Rated', value: 'models:metrics.weightedRating:desc' },
+          { label: 'Most downloaded', value: 'models:metrics.downloadCount:desc' },
+          { label: 'Most Liked', value: 'models:metrics.favoriteCount:desc' },
+          { label: 'Most discussed', value: 'models:metrics.commentCount:desc' },
+          { label: 'Newest', value: 'models:createdAt:desc' },
```

Comment:
Let's keep word casing consistency in the labels 😛 
```suggestion
          { label: 'Highest rated', value: 'models:metrics.weightedRating:desc' },
          { label: 'Most downloaded', value: 'models:metrics.downloadCount:desc' },
          { label: 'Most liked', value: 'models:metrics.favoriteCount:desc' },
          { label: 'Most discussed', value: 'models:metrics.commentCount:desc' },
          { label: 'Newest', value: 'models:createdAt:desc' },
```